src/lib/deployWorkers.ts

```diff
@@ -659,65 +655,55 @@ export const ensureAquaFileWithWorkerInfo = async (
   workersConfig: WorkersConfigReadonly,
   fluenceConfig: FluenceConfigReadonly
 ) => {
-  const allWorkerNames = Object.keys(fluenceConfig.workers ?? {});
-
-  const workerNameDealsNills = allWorkerNames.reduce<
-    Record<string, ReturnType<typeof makeOptional<Deals[string]>>>
-  >((acc, workerName) => {
-    acc[workerName] = makeOptional(null, DEAL_TYPE);
-    return acc;
-  }, {});
+  const dealWorkers = Object.fromEntries(
+    Object.entries({ ...fluenceConfig.workers, ...workersConfig.deals }).map(
+      ([workerName, info]) => {
+        const key = workerName;
+        // if worker was deployed put deal info, otherwise put null
+        const maybeDeal = "dealId" in info ? info : null;
+        const value = makeOptional(maybeDeal, DEAL_TYPE);
+        return [key, value];
+      }
+    )
+  );
 
-  const workerDealOptionals = Object.entries(workersConfig.deals ?? {}).reduce<
-    Record<string, ReturnType<typeof makeOptional<Deals[string]>>>
-  >(
-    (acc, [workerName, deal]) => {
-      acc[workerName] = makeOptional<Deals[string]>(deal, DEAL_TYPE);
-      return acc;
-    },
-    { ...workerNameDealsNills }
+  const directHostingWorkers = Object.fromEntries(
+    Object.entries({ ...fluenceConfig.workers, ...workersConfig.hosts }).map(
+      ([workerName, info]) => {
+        const key = workerName;
+        // if worker was deployed put hosts info, otherwise put null
+        const maybeHost = "relayId" in info ? info : null;
+        const value = makeOptional(maybeHost, HOSTS_TYPE);
+        return [key, value];
```

Comment:
it can but currently it's set up in such a way that you will access it in one place

Reply:
deployment won't change but access to what's been deployed will, because I think we will hardcode worker info in worker syntax instead of using arbitrary data structures

About deployment to several peers - this info is available still but under different name. `installation_spells` - it's an array. Completely unclear name, I agree. I will change it

Reply:
I'm completely lost :D